api/src/main/java/org/eclipse/microprofile/jwt/algorithm/SignatureAlgorithm.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2016-2017 Contributors to the Eclipse Foundation
+ *
+ *  See the NOTICE file(s) distributed with this work for additional
+ *  information regarding copyright ownership.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  You may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.eclipse.microprofile.jwt.algorithm;
+
+/**
+ * JWT JSON Web Signature Algorithms.
+ * 
+ * @see <a href="https://tools.ietf.org/html/rfc7518#section-3">https://tools.ietf.org/html/rfc7518#section-3</a>
+ */
+public enum SignatureAlgorithm {
+    RS256,
+    RS384,
```

Comment:
What is the purpose of the different signing algorithms when in the JWT Spec itself only RS256 is mentioned?

Reply:
@rdebusscher Because the JWT world is not only about MP JWT servers (same as the JWT tokens are not always issued with `upn` and `groups`) and it is in the interests of MP JWT IMHO for it to be usable by a larger developer audience. 
For example, even an ultra secure MP JWT server may want `RS512`. And we have a practical issue with one of Amazon services repackaging an `RS256` signed token with `ES256` as reported by our user.

Reply:
Assume the open spec as you propose, then the list of algorithms is missing

PS256, PS384, PS512, ES256K, and EdDSA (and possibly missing others)